src/pages/resources/recipes/file-operation.md

```diff
@@ -0,0 +1,317 @@
+        }
+    }
+}
+```
+
+#### manifest
+```json
+{
+    "requiredPermissions": {
+        "localFileSystem": "fullAccess"
+    }
+}
+```
+
+<br></br>
+
+**User's choice of location**
+
+These APIs are particularly handy when you want to request the user to select a folder/file location of their choice. For example, the below code snippet using `getFileForOpening` and `getFileForSaving` methods presents a file picker to the user to choose from.
+
+<CodeBlock slots="heading, code" repeat="2" languages="JSON, JavaScript" />
+
+#### JavaScript
+```js
+const fsProvider = require('uxp').storage.localFileSystem;
+
+async function foo() {
+    // Ask user to select a file. Show their 'Desktop' as the default folder.
+    if (fsProvider.isFileSystemProvider) {
+        const { domains, fileTypes } = require('uxp').storage;
+
+        try {
+                const file = await fsProvider.getFileForOpening({ initialDomain: domains.userDesktop, types: fileTypes.text });
+            if (!file) {
+                console.error("Something went wrong.");
+                return;
+            }
+            const text = await file.read();
+            console.log(`File content: ${text}`);
+        } catch (err) {
+            console.error(err);
+        }
+    }
+}
+
+async function bar() {
+    // Ask user to select a location to save a file
+    if (fsProvider.isFileSystemProvider) {
+        try {
+            const file = await fsProvider.getFileForSaving("sample.txt", { types: ["txt"] });
+            if (!file) {
+                console.error("Something went wrong.");
+                return;
+            }
+            await file.write("UXP saved sample file.");
+        } catch (err) {
+            console.error(err);
+        }        
+    }
+}
+```
+
+#### manifest
+```json
+{
+    "requiredPermissions": {
+        "localFileSystem": "request"
+    }
+}
+```
+**Save user's choice of location**
+
+If you would like to remember user's choice for an extended period of time, you can do it with the help of a token. The example below shows the essence of this usage but you should ideally save these tokens in the storage (more details covered in [Storage](./storage.md) section) for later use.
+
+```js
+const fsProvider = require('uxp').storage.localFileSystem;
+
+const { domains, fileTypes } = require('uxp').storage;
+const entry = await fsProvider.getFileForOpening({ initialDomain: domains.userDesktop, types: fileTypes.text });
+const token = await fsProvider.createPersistentToken(entry); // store for future use
+
+async function readFileUsingTokensInLocalFileSystem() {
+    // In the future, access the entry using the token directly
+    try {
+        const file = await fsProvider.getEntryForPersistentToken(token);
+        const text = await file.read();
+        console.log(`File content: ${text}`);
+    } catch (err) {
+        console.error(err);
+    }
+}
+```
+
+//TODO
+Q: What is the difference between persistent and session token?
```

Comment:
@kareemmyk

Reply:
What's DB?

Reply:
3P developers are not aware of levelDB. Can we explain this in a different way? For how long can a token remain in "session"? Or when will the session token be wiped out?

Reply:
plugins : Session is the time from the load to unload. There are 2 uses cases for plugin load

Installed Plugins : Launch of app till Quit of app
UDT Plugins: Load to unload
scripts :
Every run is a different session.